matchms/importing/parsing_utils.py

```diff
@@ -0,0 +1,86 @@
+"""Helper functions for parsing metadata.
+"""
+from typing import Any
+from typing import Union
+
+
+def find_by_key(data: Union[list, dict], target: str) -> Any:
+    """Helper function to return entries from nested list/dictionary.
+
+    Parameters
+    ----------
+    data:
+        Nested dictionary or list in which entry should be searched.
+    target:
+        Name of field to search for in data.
+    """
+    if hasattr(data, "items"):
+        for key, value in data.items():
+            if key == target:
+                yield value
+            if isinstance(value, dict):
+                yield from find_by_key(value, target)
+            elif isinstance(value, list):
+                for val in value:
+                    yield from find_by_key(val, target)
+
+    elif isinstance(data, list):
+        for subdata in data:
+            yield from find_by_key(subdata, target)
+
+
+def parse_mzml_mzxml_metadata(spectrum_dict: dict) -> dict:
+    """Parse relevant mzml (or mzxml) metadata entries.
+
+    Parameters:
+    -----------
+    spectrum_dict:
+        Spectrum dictionary containing metadata fields. Metadata parsing may easily
+        break when field key names vary. The following metadata information is considered
+        here:
+        precursor_mz, searched for in:
+        -->"precursor"/"precursorMz"--> ... --> "selected ion m/z"/"precursorMz"
+        charge, searched for in:
+        --> "charge state"/"polarity"
+        title, searched for in "spectrum title"
+        scan_start_time, searched for in "scan start time"
+        retention_time, searched for in "retentionTime"
```

Comment:
This renders as a big wrapped text, use bullets to separate for easier reading
```suggestion

        - precursor_mz, searched for in:
            -->"precursor"/"precursorMz"--> ... --> "selected ion m/z"/"precursorMz"
        - charge, searched for in:
            --> "charge state"/"polarity"
        - title, searched for in "spectrum title"
        - scan_start_time, searched for in "scan start time"
        - retention_time, searched for in "retentionTime"
```

Reply:
Nice, thanks!